pkg/asset/installconfig/aws/subnet.go

```diff
@@ -99,38 +139,59 @@ func subnets(ctx context.Context, session *session.Session, region string, ids [
 		},
 	)
 	if err != nil {
-		return vpc, nil, nil, errors.Wrap(err, "describing route tables")
+		return subnetGroups, errors.Wrap(err, "describing route tables")
+	}
+
+	azs, err := client.DescribeAvailabilityZonesWithContext(ctx, &ec2.DescribeAvailabilityZonesInput{ZoneNames: zoneNames})
+	if err != nil {
+		return subnetGroups, errors.Wrap(err, "describing availability zones")
+	}
+	for _, az := range azs.AvailabilityZones {
+		availabilityZones[*az.ZoneName] = az
 	}
 
 	publicOnlySubnets := os.Getenv("OPENSHIFT_INSTALL_AWS_PUBLIC_ONLY") != ""
 
 	for _, id := range ids {
 		meta, ok := metas[id]
 		if !ok {
-			return vpc, nil, nil, errors.Errorf("failed to find %s", id)
+			return subnetGroups, errors.Errorf("failed to find %s", id)
 		}
 
 		isPublic, err := isSubnetPublic(routeTables, id)
 		if err != nil {
-			return vpc, nil, nil, err
+			return subnetGroups, err
 		}
-		if isPublic {
-			public[id] = meta
-		} else {
-			private[id] = meta
+		meta.Public = isPublic
+		meta.ZoneType = *availabilityZones[meta.Zone].ZoneType
+		meta.ZoneGroup = *availabilityZones[meta.Zone].GroupName
+
+		// AWS Local Zones are grouped as Edge subnets
+		if meta.ZoneType == localZoneType {
+			// Local Zones is supported only in Public subnets
+			if !meta.Public {
+				return subnetGroups, errors.Errorf("Local Zones subnets must be associated with public route tables: subnet %s from availability zone %s[%s] is public[%v]", id, meta.Zone, meta.ZoneType, meta.Public)
```

Comment:
Do we want to return here or skip adding this subnet's metadata to subnetGroups.Edge and continue with  processing the next subnet id?

Reply:
Yes, return here.
> Local Zones is supported only in Public subnets

This is what we are proposing and testing for Phase-1.